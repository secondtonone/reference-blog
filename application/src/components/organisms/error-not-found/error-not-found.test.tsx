import renderer from 'react-test-renderer';
import Link from 'next/link';
import ErrorNotFound from '.';

describe('Match snapshots ErrorNotFound', () => {
  it('with heading', () => {
    const tree = renderer.create(
      <ErrorNotFound
        heading="Heading"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with caption', () => {
    const tree = renderer.create(
      <ErrorNotFound
        heading="Heading"
        caption="We can't find the page you're looking for."
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with content', () => {
    const tree = renderer.create(
      <ErrorNotFound
        heading="Heading"
        caption="We can't find the page you're looking for."
      >
        You can visit our&nbsp;
        <Link href="/blog">
          <a data-test="error-link-main">homepage</a>
        </Link>
        {' '}
        or contact our
        {' '}
        <Link href="https://www.example.com/company/contacts/">
          <a data-test="error-link-support" target="_blank">support team</a>
        </Link>
        .
      </ErrorNotFound>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
