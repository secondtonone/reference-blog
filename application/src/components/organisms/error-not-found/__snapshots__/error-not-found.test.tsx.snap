// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ErrorNotFound with caption 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 fiLzXF"
>
  <h1
    className="text-content__TextStyled-sc-1stkrsz-0 dRRiqw error-not-found__HeadingStyled-sc-1f8qoay-1 jlRpDg"
    data-test="error-heading"
  >
    Heading
  </h1>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 kuJVxh error-not-found__MessageStyled-sc-1f8qoay-0 bCiwPl"
  >
    <p
      className="text-content__TextStyled-sc-1stkrsz-0 bUJWgm error-not-found__TextContentStyled-sc-1f8qoay-2 dpYmzX"
      data-test="error-caption"
    >
      We can't find the page you're looking for.
    </p>
  </div>
</div>
`;

exports[`Match snapshots ErrorNotFound with content 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 fiLzXF"
>
  <h1
    className="text-content__TextStyled-sc-1stkrsz-0 dRRiqw error-not-found__HeadingStyled-sc-1f8qoay-1 jlRpDg"
    data-test="error-heading"
  >
    Heading
  </h1>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 kuJVxh error-not-found__MessageStyled-sc-1f8qoay-0 bCiwPl"
  >
    <p
      className="text-content__TextStyled-sc-1stkrsz-0 bUJWgm error-not-found__TextContentStyled-sc-1f8qoay-2 dpYmzX"
      data-test="error-caption"
    >
      We can't find the page you're looking for.
    </p>
    <p
      className="text-content__TextStyled-sc-1stkrsz-0 bUJWgm error-not-found__TextContentStyled-sc-1f8qoay-2 dpYmzX"
      data-test="error-text"
    >
      You can visit ourÂ 
      <a
        data-test="error-link-main"
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        homepage
      </a>
       
      or contact our
       
      <a
        data-test="error-link-support"
        href="https://www.example.com/company/contacts/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        target="_blank"
      >
        support team
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`Match snapshots ErrorNotFound with heading 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 fiLzXF"
>
  <h1
    className="text-content__TextStyled-sc-1stkrsz-0 dRRiqw error-not-found__HeadingStyled-sc-1f8qoay-1 jlRpDg"
    data-test="error-heading"
  >
    Heading
  </h1>
</div>
`;
